src/LAND/LAND.router.ts

```diff
@@ -0,0 +1,194 @@
+import { env } from 'decentraland-commons'
+import { server } from 'decentraland-server'
+import FormData from 'form-data'
+import { Request } from 'express'
+//@ts-ignore
+import * as contentHash from 'content-hash'
+import fetch, { Response as FetchResponse } from 'node-fetch'
+import { HTTPError, STATUS_CODES } from '../common/HTTPError'
+import { Router } from '../common/Router'
+import { getCID } from '../utils/cid'
+import {
+  GetRedirectionHashesResponse,
+  UploadRedirectionResponse,
+} from './LAND.types'
+
+export class LANDRouter extends Router {
+  mount() {
+    this.router.get(
+      '/lands/redirection/hashes',
+      server.handleRequest(this.getRedirectionHashes)
+    )
+
+    this.router.post(
+      '/lands/:coords/redirection',
+      server.handleRequest(this.uploadRedirection)
+    )
+  }
+
+  private uploadRedirection = async (
+    req: Request
+  ): Promise<UploadRedirectionResponse> => {
+    const { url, projectId, apiKey } = this.getEnvs()
+
+    const coords = server.extractFromReq(req, 'coords')
+
+    this.validateCoords(coords)
+
+    const locale = req.headers['accept-language']
+
+    const redirectionFile = this.generateRedirectionFile(coords, locale)
+
+    const formData = new FormData()
+
+    formData.append('blob', redirectionFile, 'index.html')
+
+    let result: FetchResponse
+
+    try {
+      result = await fetch(url + '/api/v0/add?pin=false', {
+        method: 'post',
+        body: formData,
+        headers: {
+          Authorization: `Basic ${Buffer.from(
+            `${projectId}:${apiKey}`
+          ).toString('base64')}`,
+        },
+      })
+    } catch (e) {
+      throw new HTTPError(
+        'Failed to upload file to IPFS # Request',
+        {},
+        STATUS_CODES.error
+      )
+    }
+
+    if (!result.ok) {
+      throw new HTTPError(
+        'Failed to upload file to IPFS # Response',
```

Comment:
What if we add more information to the response?
```suggestion
        'Failed to upload file to IPFS # Response, the IPFS server responded with a x error',
```

Reply:
Changed the message